### PR TITLE
Fix bug that causes custom context tag settings to be overwritten

### PR DESCRIPTION
**Pull request by lmillott**
Currently if you set a custom user Id or session Id it is overwritten by ServerRequestParser.ts. If there is already a setting for that context key, it should not be overwritten.

Additionally fixes an issue caused by the recent change to set the ikeyhash as a header on the response. It throws if the response has already been sent. Bug was introduced in this changeset: https://github.com/Microsoft/ApplicationInsights-node.js/commit/0a63c3fd3a959c88ba6751d53e633a14fd769f9a

Also fixed a couple build errors on windows using ts 2.0.3

### CODE REVIEW

File: Tests/AutoCollection/ServerRequestParser.tests.ts [hunk 1]
```diff
@@ -95,4 +95,40 @@ describe("AutoCollection/ServerRequestParser", () => {
             assert.equal(requestData.baseData.url, "http://bing.com/");
         });
     });
+// newTags[ServerRequestParser.keys.locationIp] = tags[ServerRequestParser.keys.locationIp] || this._getIp();
```
Comment (kamilszostak) [inline Tests/AutoCollection/ServerRequestParser.tests.ts]: remove commented out code

File: Library/Util.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ class Util {
             var cookieName = name + "=";
             var cookies = cookie.split(";");
             for (var i = 0; i < cookies.length; i++) {
-                var cookie = cookies[i];
+                var cookie: any = cookies[i];
```
Comment (kamilszostak) [inline Library/Util.ts]: why do you need to cast it to `any`?

File: Tests/AutoCollection/ServerRequestParser.tests.ts (hunk 1 above)
Comment (lmillott) [inline Tests/AutoCollection/ServerRequestParser.tests.ts]: thanks

File: Library/Util.ts (hunk 2 above)
Comment (lmillott) [inline Library/Util.ts]: it's causing a build error for me. Probably different typescript version. I've removed it

File: AutoCollection/RequestParser.ts [hunk 3]
```diff
@@ -30,7 +30,7 @@ abstract class RequestParser {
             } else if (error instanceof Error) {
                 properties["error"] = error.message;
             } else if (typeof error === "object") {
-                for (var key in error) {
+                for (var key in <any>error) {
```
Comment (lmillott) [inline AutoCollection/RequestParser.ts]: it causes a build error for me. Probably different typescript version. I've removed it

**Pull request merged**